src/Subscriptions.ts

```diff
@@ -68,28 +57,46 @@ export function handleFeeAmountChanged(event: FeeAmountChanged): void {
   subscriptions.save()
 }
 
-export function handlePrePaymentPeriodsChanged(event: PrePaymentPeriodsChanged): void {
-  let subscriptions = SubscriptionModule.load(event.address.toHex())
-  subscriptions.prePaymentPeriods = event.params.currentPrePaymentPeriods
-  subscriptions.save()
-}
-
 export function handleGovernorSharePctChanged(event: GovernorSharePctChanged): void {
   let subscriptions = SubscriptionModule.load(event.address.toHex())
   subscriptions.governorSharePct = BigInt.fromI32(event.params.currentGovernorSharePct)
   subscriptions.save()
 }
 
-export function handleLatePaymentPenaltyPctChanged(event: LatePaymentPenaltyPctChanged): void {
-  let subscriptions = SubscriptionModule.load(event.address.toHex())
-  subscriptions.latePaymentPenaltyPct = BigInt.fromI32(event.params.currentLatePaymentPenaltyPct)
-  subscriptions.save()
+export function handleAppFeeSet(event: AppFeeSet): void {
+  let appFee = loadOrCreateAppFee(event.params.appId)
+  appFee.isSet = true
+  appFee.amount = event.params.amount
+  appFee.instance = event.address.toHex()
+  appFee.save()
 }
 
-export function handleResumePenaltiesChanged(event: ResumePenaltiesChanged): void {
-  let subscriptions = SubscriptionModule.load(event.address.toHex())
-  subscriptions.resumePrePaidPeriods = event.params.currentResumePrePaidPeriods
-  subscriptions.save()
+export function handleAppFeeUnset(event: AppFeeUnset): void {
+  let appFee = AppFee.load(event.params.appId.toString())
+
+  if (appFee === null) {
+    return
+  }
+  appFee.isSet = false
+  appFee.amount = BigInt.fromI32(0)
+  appFee.save()
+}
+
+export function handleAppFeePaid(event: AppFeePaid): void {
+  let appFee = AppFee.load(event.params.appId.toString())
+
+  if (appFee === null) {
```

Comment:
This should never occur right? I think I'd use `loadOrCreate` here as well

Reply:
No, it should never occur, I added this as an extra safety measure. I don’t think I would use `loadOrCreate` here, because if for some weird reason `appFee` was null, we wouldn’t have the amount. But no strong opinions.